.github/workflows/publish.yml

```diff
@@ -11,34 +11,73 @@ on:
     paths:
       - "src/**"
       - ".github/workflows/publish.yml"
+  repository_dispatch:
+    types: [create-release]
+    client_payload:
+      release_tag:
+        required: true
 
 jobs:
+  create_release:
+    if: ${{ github.event_name == 'repository_dispatch' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: Brightspace/third-party-actions@actions/checkout
+      - name: Create release
+        env:
+          RELEASE_TAG: ${{ github.event.client_payload.release_tag }}
+        run: |
+          gh release create "$RELEASE_TAG" --draft `
+            --title "Release $RELEASE_TAG" `
+            --notes "Release notes for version $RELEASE_TAG"
+
   build:
     strategy:
       matrix:
         include:
           - machine: windows-latest
             platform: win
+            file_name: D2L.Bmx.exe
           - machine: macos-latest
             platform: osx
+            file_name: D2L.Bmx
     runs-on: ${{ matrix.machine }}
     timeout-minutes: 10
+    env:
+      PLATFORM: ${{ matrix.platform }}
     steps:
       - name: checkout
         uses: Brightspace/third-party-actions@actions/checkout
       - name: publish
         shell: pwsh
+        working-directory: src/D2L.Bmx
+        run: |
+          dotnet publish -r $env:PLATFORM-x64 -c Release -o build
+      - name: upload build
```

Comment:
I thought to have this step as a reusable action but it just ended up being even more code lol